add doc for cloudwatch container insights (#103)
aws-otel/aws-otel.github.io: src/docs/getting-started/cloudwatch-container-insights.mdx

---
title: 'Using CloudWatch Metrics with AWS Distro for OpenTelemetry'
description:
    CloudWatch Container Insights collects, aggregates, and summarize metrics and logs from your containerized applications and microservices. 

>     In this tutorial, we will walk through how to enable CloudWatch Container Insights with AWS OTel Collector for an EKS cluster. 

Author: Will change accordingly.

path: '/docs/getting-started/cloudwatch-container-insight'
---

import { Link } from "gatsby"
import SectionSeparator from "components/MdxSectionSeparator/sectionSeparator.jsx"
import imgLogGroup from "assets/img/docs/gettingStarted/containerInsights/log-group.png"
import imgPodMetrics from "assets/img/docs/gettingStarted/containerInsights/pod-metrics.png"

[CloudWatch Container Insights](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/ContainerInsights.html) collect, aggregate, 
and summarize metrics and logs from your containerized applications and microservices. Data are collected as as performance log events 
using [embedded metric format](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch_Embedded_Metric_Format.html). 
These performance log events are entries that use a structured JSON schema that enables high-cardinality data to be ingested and stored at scale. 

> From this data, CloudWatch creates aggregated metrics at the cluster, node, pod, task, and service level as CloudWatch metrics. 

Reviewer: ```suggestion
Amazon CloudWatch can creates the aggregated CloudWatch metrics for the cluster, node, pod, task, and service level from the received EMF data.
```

Author: Will change accordingly.

> Exsting CloudWatch Container Insights relies on ECS Agent and CloudWatch Agent to collect infrastructure metrics for many resources such as

Reviewer: ```suggestion
CloudWatch Container Insights has been supported by ECS Agent and CloudWatch Agent to collect infrastructure metrics for many resources such as
```

Author: Will change accordingly.

such as CPU, memory, disk, and network. To migrate existing customers to use AWS Distro for OpenTelemetry, we are currently enhancing the 
AWS OTel Collector to support the same CloudWatch Container Insights experience for the following platforms:  
  * Amazon ECS 
  * Amazon EKS
  * Kubernetes platforms on Amazon EC2

For Amazon ECS, the cluster and service level metrics are already supported ([see the public AWS doc for details](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/deploy-container-insights-ECS-adot.html))
and the support for instance level metrics is working in progress. For Amazon EKS and Kubernetes platforms on Amazon on EC2, all the infrastructure metrics are supported. 
In this tutorial, we will walk through how to enable CloudWatch Container Insights with AWS OTel Collector for an EKS cluster. 

<SectionSeparator />

## Getting Started

> To use AWS OTel Collector to collect infrastructure metrics for your cluster, we need to [make sure all the prerequisites are satisfied.](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/Container-Insights-prerequisites.html)

Reviewer: ```suggestion
To use AWS OTel Collector to collect infrastructure metrics for your service cluster, we need to make sure [all the prerequisites](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/Container-Insights-prerequisites.html) are satisfied.
```

Author: Will change accordingly.

to confirm the AWS OTel Collector is running:
```
kubectl get pods -l name=aws-otel-eks-ci -n aws-otel-eks
```
If the results include multiple pods (one for each cluster node) in the Running state, the collector is running and collecting metrics from the cluster.
The AWS OTel Collector creates a log group named `/aws/containerinsights/MyCluster/performance` and sends the performance log events to this log group.
Each collector pod on a cluster node will publish logs to a log stream with the name of the cluster node. In the screenshot, three log streams are present
under the log group `/aws/containerinsights/ci-demo/performance` and each corresponds to one cluster node:
<img src={imgLogGroup} alt="Diagram" style="margin: 30px 0;" />

The metrics that Container Insights collects are also available in CloudWatch automatic dashboards as different resource types, such as 
`EKS Clusters`, `EKS Namespaces`, `EKS Nodes`, `EKS Services`, and `EKS Pods`. Here is a screenshot for the pod level metrics for 
a cluster named `ci-demo`:
<img src={imgPodMetrics} alt="Diagram" style="margin: 30px 0;" /> 

> ## Configurations to support CloudWatch Container Insights for EKS

Reviewer: ```suggestion
## Configurations to support CloudWatch Container Insights for EKS EC2
```

Author: Will change accordingly.
